### PR TITLE
Fix GCS system tests

### PR DESCRIPTION
**Pull request by deedmitrij**
- Add '**/tmp/**' path to names of creating files
- Add separate methods and fixtures for creating\deleting required files
- Add `resource` key which disables uniform bucket level access to **GCSCreateBucketOperator**

### CODE REVIEW

File: airflow/providers/google/cloud/example_dags/example_gcs.py
Comment (potiuk) [inline airflow/providers/google/cloud/example_dags/example_gcs.py]: similarly to #19208 we should use gettempdir() here and below.

**Review by potiuk — CHANGES_REQUESTED** [review]
NIT: tmp file location.

Greate we are switching to fixtures.

File: airflow/providers/google/cloud/example_dags/example_gcs.py
Comment (deedmitrij) [reply airflow/providers/google/cloud/example_dags/example_gcs.py]: Ok, done

File: airflow/providers/google/cloud/example_dags/example_gcs.py
Comment (uranusjr) [inline airflow/providers/google/cloud/example_dags/example_gcs.py]: Please use `os.path.join` instead.

Comment (deedmitrij) [reply airflow/providers/google/cloud/example_dags/example_gcs.py]: Ok, done

**Comment by deedmitrij** [comment]
Any updates?

File: tests/providers/google/cloud/operators/test_gcs_system.py
Comment (uranusjr) [inline tests/providers/google/cloud/operators/test_gcs_system.py:27]: Can we give this fixture a more descriptive name?

**Review by uranusjr — APPROVED** [review]
Generally good to me

**Comment by mnojek** [comment]
Some checks are failing probably due to changes made in #19609

**Comment by mnojek** [comment]
Hey @uranusjr @potiuk 
Some checks are failing due to changes done in #19609. Is there a plan to fix it or maybe it's already fixed?

**Comment by potiuk** [comment]
What makes you think it's caused by #19609 ?  - (BTW: you are 57 commits behind so I suggest rebase (though we seem to have an OOM in `main` which I am looking at now, so maybe wait with teh rebase.

**Comment by potiuk** [comment]
Likely fix to the current OOM issues here: https://github.com/apache/airflow/pull/19766

**Comment by mnojek** [comment]
@potiuk I rebased the branch and rerun the tests. Still one check is failing, not sure exactly what's the cause.

**Comment by mnojek** [comment]
@potiuk @uranusjr @turbaszek
The static checks have passed. Please approve if everything's fine and merge. Thanks!

File: tests/providers/google/cloud/operators/test_gcs_system.py
Comment (mnojek) [reply tests/providers/google/cloud/operators/test_gcs_system.py:27]: Do you have any proposition? Looks fine for me, but I can change it, if you want. For example, I can change it to `gcs_helper`.

Comment (mnojek) [reply tests/providers/google/cloud/operators/test_gcs_system.py:27]: @uranusjr @deedmitrij I cannot resolve the issue cause I'm not the author of the PR. Please resolve or answer or guide me what changes should I do because it's blocking the PR to be merged.

Comment (uranusjr) [reply tests/providers/google/cloud/operators/test_gcs_system.py:27]: `gcs_helper` sounds good to me. The PR looks good to me besides this as well (and I already approved), so let’s wait for another maintainer to take a look and merge if they feel it’s good to go.

Comment (potiuk) [reply tests/providers/google/cloud/operators/test_gcs_system.py:27]: Helper is as good as it can get. I also do not like general "Util, Helper, Manager, Processor" but if you look at Airflow code we have quite a bunch of those everywhere (and as much as we fight it - there ARE cases where they are needed and actually usually in tests, I'd be far more relaxed to have those).

**Review by potiuk — APPROVED** [review]

**Pull request merged**
